### ISSUE TITLE
diagnose various name resolution errors in transcripts

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#2722]
- [ ] fix2158-1.md -- can't resolve `List.map`
- [ ] mvar.md -- can't resolve pattern `Left`
- [ ] net.md -- can't resolve `compose2`
- [ ] stm.md -- can't resolve `Nat`, `TVar`, `io2.IO`, `Optional`
- [ ] thread.md -- can't resolve `io2.IO`, `Result`
- [ ] tls.md -- can't resolve pattern `Left`

**Comment by ChrisPenner** [comment]
Looks like these are failing on trunk as well if you run them manually, e.g. `stack exec transcripts -- stm.md`.

They all depend on definitions in base, but don't ever actually import or setup base.

**Comment by ChrisPenner** [comment]
Dug into this a bit, looks like the actual failures in these transcripts are resolved by #2730 , but what probably happened is that you looked at the failing transcripts in the CI printout, then ran them one by one using `stack exec transcripts -- transcript-name`, which caused transcripts which depend on base to fail with missing identifiers due to #2738 ...

![image](https://user-images.githubusercontent.com/6439644/145065068-0a03b0f8-9bb9-47ce-b44f-32b935f91c95.png)

TLDR; #2730 fixes these transcripts (I merged your branch with the fix in #2738 and tested it), but I've made #2738 to fix the running of individual base transcripts so we won't hit this red-herring again 😄
